datafusion/common/src/dfschema.rs

```diff
@@ -564,6 +564,7 @@ impl DFSchema {
     }
 
     /// Check to see if fields in 2 Arrow schemas are compatible
+    #[deprecated(since = "47.0.0", note = "This method is no longer used")]
```

Comment:
Do we have a replaceable method for this one?

Reply:
I believe we have a similar API already:

logically_equivalent_names_and_types